src/Sylius/Bundle/ApiBundle/spec/Validator/CatalogPromotion/FixedDiscountActionValidatorSpec.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\ApiBundle\Validator\CatalogPromotion;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Bundle\ApiBundle\SectionResolver\AdminApiSection;
+use Sylius\Bundle\ApiBundle\SectionResolver\ShopApiSection;
+use Sylius\Bundle\CoreBundle\SectionResolver\SectionProviderInterface;
+use Sylius\Bundle\PromotionBundle\Validator\CatalogPromotionAction\ActionValidatorInterface;
+use Sylius\Bundle\PromotionBundle\Validator\Constraints\CatalogPromotionAction;
+use Sylius\Bundle\ShopBundle\SectionResolver\ShopSection;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Promotion\Model\CatalogPromotionActionInterface;
+use Symfony\Component\Validator\Context\ExecutionContextInterface;
+use Symfony\Component\Validator\Violation\ConstraintViolationBuilderInterface;
+
+final class FixedDiscountActionValidatorSpec extends ObjectBehavior
```

Comment:
If I see correctly, we didn't cover the case with two channel configurations - right now, it will add only one violation on configuration level. If we want to keep this behaviour, I would vote for adding this case. Otherwise, we need to test it and adjust code ;)

Reply:
👍 I would leave it as it is right now, we can adjust it later